internal/commands/vaultsecrets/secrets/read.go

```diff
@@ -0,0 +1,145 @@
+	"github.com/hashicorp/hcp/internal/pkg/iostreams"
+	"github.com/hashicorp/hcp/internal/pkg/profile"
+)
+
+func NewCmdRead(ctx *cmd.Context, runF func(*ReadOpts) error) *cmd.Command {
+	opts := &ReadOpts{
+		Ctx:           ctx.ShutdownCtx,
+		Profile:       ctx.Profile,
+		IO:            ctx.IO,
+		Output:        ctx.Output,
+		PreviewClient: preview_secret_service.New(ctx.HCP, nil),
+		Client:        secret_service.New(ctx.HCP, nil),
+	}
+
+	cmd := &cmd.Command{
+		Name:      "read",
+		ShortHelp: "Read a static secret.",
+		LongHelp: heredoc.New(ctx.IO).Must(`
+		The {{ template "mdCodeOrBold" "hcp vault-secrets secrets read" }} command reads a static secret from the Vault Secrets application.
+		`),
+		Examples: []cmd.Example{
+			{
+				Preamble: `Read secret's metadata:`,
+				Command: heredoc.New(ctx.IO, heredoc.WithPreserveNewlines()).Must(`
+				$ hcp vault-secrets secret read "test_secret"
+				`),
+			},
+			{
+				Preamble: `Read plaintext secret:`,
+				Command: heredoc.New(ctx.IO, heredoc.WithPreserveNewlines()).Must(`
+				$ hcp vault-secrets secret read "test_secret" --plaintext
+				`),
+			},
+		},
+		Args: cmd.PositionalArguments{
+			Args: []cmd.PositionalArgument{
+				{
+					Name:          "NAME",
+					Documentation: "The name of the secret to read.",
+				},
+			},
+		},
+		Flags: cmd.Flags{
+			Local: []*cmd.Flag{
+				{
+					Name:          "plaintext",
+					Description:   "Display the secret's value in plaintext.",
+					Value:         flagvalue.Simple(false, &opts.OpenSecret),
+					IsBooleanFlag: true,
+				},
+			},
+		},
+		RunF: func(c *cmd.Command, args []string) error {
+			opts.AppName = appName
+			opts.SecretName = args[0]
+
+			if runF != nil {
+				return runF(opts)
+			}
+			return readRun(opts)
+		},
+	}
+
+	return cmd
+}
+
+type ReadOpts struct {
+	Ctx     context.Context
+	Profile *profile.Profile
+	IO      iostreams.IOStreams
+	Output  *format.Outputter
+
+	AppName       string
+	SecretName    string
+	OpenSecret    bool
+	PreviewClient preview_secret_service.ClientService
+	Client        secret_service.ClientService
+}
+
+func readRun(opts *ReadOpts) error {
+	if opts.OpenSecret {
+		return runOpenAppSecret(opts)
+	}
+
+	req := secret_service.NewGetAppSecretParamsWithContext(opts.Ctx)
+	req.LocationOrganizationID = opts.Profile.OrganizationID
+	req.LocationProjectID = opts.Profile.ProjectID
+	req.AppName = opts.AppName
+	req.SecretName = opts.SecretName
+
+	resp, err := opts.Client.GetAppSecret(req, nil)
+	if err != nil {
+		return fmt.Errorf("failed to read the secret %q: %w", opts.SecretName, err)
+	}
+	return opts.Output.Display(newDisplayer(true).Secrets(resp.Payload.Secret))
+}
+
+func runOpenAppSecret(opts *ReadOpts) error {
+	// Table format doesn't handle large secret values well. So until that is fixed,
+	// we should not allow table format for this command.
+	if opts.Output.GetFormat() == format.Table {
+		return fmt.Errorf("table format is not supported for this command")
```

Comment:
Let's not do this. If the table output isn't well formatted, the user can specify an alternative format.

Reply:
Trying to understand - we don't handle this well today, so why put it out there?

Reply:
After discussing, we think it ok to add it as for shorter creds this won't be a problem.